Source/Core/DolphinQt2/Translation.cpp

```diff
@@ -0,0 +1,239 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "DolphinQt2/Translation.h"
+
+#include <QApplication>
+#include <QLocale>
+#include <QMessageBox>
+#include <QTranslator>
+#include <algorithm>
+#include <cstring>
+#include <iterator>
+
+#include "Common/File.h"
+#include "Common/FileUtil.h"
+#include "Common/Logging/Log.h"
+#include "Common/MsgHandler.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+
+static constexpr u32 MO_MAGIC_NUMBER = 0x950412de;
+
+static u16 ReadU16(const char* data)
+{
+  u16 value;
+  std::memcpy(&value, data, 2);
+  return value;
+}
+
+static u32 ReadU32(const char* data)
+{
+  u32 value;
+  std::memcpy(&value, data, 4);
+  return value;
+}
+
+class MoIterator : public std::iterator<std::random_access_iterator_tag,  // iterator_category
```

Comment:
`std::iterator` is deprecated as [P0174](http://www.open-std.org/jtc1/sc22/wg21/docs/papers/2016/p0174r2.html#2.1) was voted into C++17. You need to provide the type definitions yourself.

This also doesn't implement all the mandated operations that the [RandomAccessIterator](http://en.cppreference.com/w/cpp/concept/RandomAccessIterator) concept requires of it.

Reply:
Okay, I replaced `std::iterator` with the typedefs (that's all it did? Seems like it was unnecessary to have in the first place).

I'm hesitant to add the rest of the RandomAccessIterator requirements, though, since it's 1) a ton of boilerplate that 2) won't get tested, since it's not used by `std::lower_bound`. Is fulfilling every requirement necessary? If another STL implementation does use them, it's probably better to raise a compilation error at that point than to have untested code.

Reply:
> Is fulfilling every requirement necessary?

Yes (they are *requirements* after all). Otherwise it's not a compliant random access iterator. You could alternatively modify it to follow the [BidirectionalIterator](http://en.cppreference.com/w/cpp/concept/BidirectionalIterator) concept if you don't want to do the testing to ensure it's a compliant RandomAccessIterator.